### ISSUE TITLE
Allow parameters of SplitOperatorAnsatz to be coefficients of entire Hamiltonian terms

### ISSUE THREAD

**Issue by kevinsung** [quantumlib/OpenFermion-Cirq#282]
Currently, the parameters of SplitOperatorAnsatz are angles of individual gates. However, another natural choice is to have the parameters actually be the coefficients of entire Hamiltonian terms. For instance, the Hamiltonian is the sum of two terms, T + V, and we want the ansatz to be something like exp(-iaT) exp(-ibV), where a and b are the parameters. When you compile this down to gates, it turns out that a and b will appear as coefficients in the angles of the gates, so this seems to require Symbols to support arithmetic if we want to use them.

In short, this issue is blocked by Symbols not supporting multiplication.
